File: DRIVER_MANAGEMENT_PROJECT_BACKEND/src/main/java/com/eren/DRIVER_MANAGEMENT_PROJECT_BACKEND/entity/enums/Tasks.java
package com.eren.DRIVER_MANAGEMENT_PROJECT_BACKEND.entity.enums;

/**
 * Represents the status of a task within the system.
 *
 * <p>This enum defines various stages that a task can be in during its lifecycle,
 * such as planned, in progress, completed, or cancelled.</p>
 */
public enum Tasks {

    /** The task has been completed successfully. */
    COMPLETED,

    /** The task is currently in progress. */
    IN_PROGRESS,

    /** The task has been cancelled and will not be completed. */
    CANCELLED,

    /** The task is scheduled but has not started yet. */
    PLANNED
}

File: DRIVERI_MANAGEMENT_PROJECT_BACKEND/Entities/Enums/Tasks.cs
using System.ComponentModel.DataAnnotations;
namespace Entities.Enums
{
    public enum Tasks
    {
        [Display(Name ="Completed")]
        COMPLETED = 0,
        [Display(Name = "In Progress")]
        IN_PROGRESS = 1,
        [Display(Name = "Cancelled")]
        CANCELLED = 2,
        [Display(Name = "Planned")]
        PLANNED = 3,
    }
}
